### ISSUE TITLE
Red Scoreboard Numbers not hidden on 1.19.3 server and 1.20+ clients

### ISSUE THREAD

**Issue by CreationMC1** [PluginBugs/Issues-ItemsAdder#3195]
### Terms

- [X] I'm using the very latest version of ItemsAdder and its dependencies.
- [X] I am sure this is a bug and it is not caused by a misconfiguration or by another plugin.
- [X] I already searched on this [Github page](https://github.com/PluginBugs/Issues-ItemsAdder/issues) to check if the same issue was already reported.
- [X] I already searched on the [plugin wiki](https://itemsadder.devs.beer/) to know if a solution is already known.
- [X] I already searched on the [forums](https://forum.devs.beer/) to check if anyone already has a solution for this.

### Discord tag (optional)

creationmc

### What happened?

The numbers on the scoreboard appear even when disabled within the configuration. Image is below as an example. 

### Steps to reproduce the issue

#1 - Load up any scoreboard.
#2 - Disable the red numbers in the items adder configuration.
#3 - Watch as the numbers don't disappear. 

### Server version

This server is running Pufferfish version git-Pufferfish-29 (MC: 1.20.2) (Implementing API version 1.20.2-R0.1-SNAPSHOT) 

### ItemsAdder Version

3.6.2c

### ProtocolLib Version

5.1.1-SNAPSHOT-673

### LoneLibs Version

1.0.31

### FULL server log

N/A

### Screenshots/Videos (you can drag and drop files or paste links)

Image: https://i.imgur.com/tVsaPBI.png

**Comment by CreationMC1** [comment]
Yes, I did @LoneDev6 - I'm unsure if that's helpful or not.

**Comment by LoneDev6** [comment]
You might have another plugin or pack interfering with text shaders used by ItemsAdder to hide the numbers.
Anyway 1.20.4 introduces a new way to hide scoreboard numbers and will be available in the next ItemsAdder update.
https://github.com/PluginBugs/Issues-ItemsAdder/issues/3205

**Comment by LoneDev6** [comment]
I confirm that the new option `hide-scoreboard-numbers` is not working because 1.19.4 doesn't implement a packet used by ItemsAdder in 1.20.4 to remove the numbers from the scoreboard without using shaders.

In the next update I will fix the shader to support for 1.20.4 too.

-----

# Tests

## 1.19.4 server and client
```yml
hide-scoreboard-numbers: false
hide-scoreboard-numbers-old-clients: true
```
Then run /iazip
Uses shaders to do that.

![image](https://github.com/PluginBugs/Issues-ItemsAdder/assets/27242001/c5d8d281-e93e-4e21-8879-a8fc3e7ab2b9)

## 1.20.4 server and client
```yml
hide-scoreboard-numbers: true
hide-scoreboard-numbers-old-clients: false
```
Doesn't use shaders to do that. No need to run /iazip.

![image](https://github.com/PluginBugs/Issues-ItemsAdder/assets/27242001/bad54908-705d-45bd-9b1a-c24a7c1c81ad)

## 1.20.4 server and 1.19.4 client
```yml
hide-scoreboard-numbers: true # Doesn't matter true/false
hide-scoreboard-numbers-old-clients: true
```

![image](https://github.com/PluginBugs/Issues-ItemsAdder/assets/27242001/67dcfcfa-caf2-4ef1-b4b3-a85b090388b0)

**Comment by ColorsASD** [comment]
Thank you for listening and that you will fix it in the near future. <3

**Comment by ColorsASD** [comment]
Clearly, the reference to `text-effect` is because it influences `scoreboard-numbers`. If you wish, you could add an additional setting related to `text-effect`, specifically one that removes the shadow from the text. This can also be achieved using color codes.

On Discord, you can find a solution, but some users complained that they don't understand why they need to disable `text-effect`, so adding a setting for it could address this concern.

Extremely useful for HUDs as it eliminates the shadow, making it look more aesthetically pleasing.
You can supplement this setting.

```yml
  force_color:
    enabled: true
    color: '#FFFEFD'
```
_(To my knowledge, this setting was created upon a user's request, as they also wanted to remove shadows.)_

[Discord link...](https://discord.com/channels/533407895010803753/1192241214678904862/1192241214678904862)

**Comment by LoneDev6** [comment]
`text-effect: true` is not required anymore to make `hide-scoreboard-numbers-old-clients: true` actually extract the needed shader.
You can now disable text effects and hide the scoreboard numbers as I made a new shader for that only.

About the hide shadow suggestion, please open a new issue report and explain exactly what you want to achieve, I will take into consideration to bundle the suggested shader.

Thanks!

**Comment by xiaozhangup** [comment]
I would like to know how this is implemented in 1.20.3+, is there any documentation or other helpful information? Much appreciated!
